internal/sync-map/map.go

```diff
@@ -0,0 +1,81 @@
+package sync_map
+
+import (
+	"fmt"
+	"sync"
+	"github.com/Capstone-auto-grader/grader-api-v2/internal/grader-task"
+)
+// A synchronized map to store Tasks
```

Comment:
comment should start with "SyncMap is ..."/"SyncMap ... ": https://golang.org/doc/effective_go.html#commentary